sys/kern/subr_intr.c

```diff
@@ -815,7 +815,7 @@ pic_create(device_t dev, intptr_t xref, u_int flags)
 	pic->pic_dev = dev;
 	pic->pic_flags = flags;
 	mtx_init(&pic->pic_child_lock, "pic child lock", NULL, MTX_SPIN);
-	SLIST_INSERT_HEAD(&pic_list, pic, pic_next);
+	STAILQ_INSERT_TAIL(&pic_list, pic, pic_next);
```

Comment:
You should at the very least have a comment mentioning that the insertion order is important, since it preserves an invariant that you apparently depend on.

Reply:
I don't depend on it.  There was a statement the proposed IPI PIC for RISC-V would depend on this though.  In truth there are so few uses of `pic_init_secondary()` that doing anything is simply a guessing game.

Since this seems so controversial I'll likely be switching to declaring the Xen entry as a separate root PIC.  This is _mildly_ abusing that functionality, but does seem a good choice as a backup solution.

Reply:
If you don't require this, then it doesn't belong in minimal set of patches for running Xen on arm64. I don't have strong opinions on the issue at hand, but since it's difficult to get consensus on intrng changes, your goal is easier to achieve if we can avoid those that aren't strictly needed.

Reply:
*I* don't depend on any invariant.  There are claims of there being invariants, but I can't verify any of those.  Notably the call I need doesn't care about ordering.

Some sort of hook into processor startup is needed.  Otherwise we end up with all Xen events on a single processor.  There is also an alternative approach which showed up _after_ this had last been updated.

Reply:
This was trying to figure out how to get hooked into `pic_init_secondary()`, #1280.  Due to the changes in how this is handled due to #1363, this no longer applies.

Still a bit hazy as to which direction to go with getting connected to `pic_init_secondary()` though.